conan/tools/cmake/file_api.py

```diff
@@ -15,19 +16,12 @@ class CMakeFileAPI(object):
     def __init__(self, conanfile):
         self._conanfile = conanfile
 
-    @property
-    def build_dir(self):
```

Comment:
Avoid public API of user-facing components as much as possible, specially those that users already have another handle (like ``self.build_folder``